File: flame/src/main/java/org/movieos/flame/models/FlameService.java
package org.movieos.flame.models;

import android.net.nsd.NsdServiceInfo;

import org.movieos.flame.utilities.ServiceLookup;

import java.util.ArrayList;

public class FlameService {

    NsdServiceInfo mService;
    ArrayList<String> mHostIdentifiers;
    ServiceLookup mServiceLookup;

    public FlameService(NsdServiceInfo s) {
        if (s == null) {
            throw new RuntimeException("service must be set");
        }
        mService = s;
        mHostIdentifiers = new ArrayList<>();
//        if (service.getInfo() != null) {
//            if (!service.getInfo().getServer().isEmpty()) {
//                hostIdentifiers.add(service.getInfo().getServer());
//            }
//            for (String address : service.getInfo().getHostAddresses()) {
//                if (!address.isEmpty()) {
//                    hostIdentifiers.add(address);
//                }
//            }
//            hostIdentifiers.add(toString());
//
//            serviceLookup = ServiceLookup.get(service.getInfo().getApplication());
//        }
    }

    public ArrayList<String> getHostIdentifiers() {
        return mHostIdentifiers;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof FlameService)) {
            return false;
        }
        return ((FlameService) o).toString().equals(toString());
    }

    public String toString() {
        return String.format("%s %s", mService.getServiceName(), mService.getServiceType());
    }

    public NsdServiceInfo getInfo() {
        return mService;
    }

    public String getIPAddress() {
        return mService.getHost().getHostAddress();
    }

    public ServiceLookup getServiceLookup() {
        return mServiceLookup;
    }

    public String getIdentifier() {
        // TODO probably not good enough
        return String.format("%s %s", mService.getServiceName(), mService.getServiceType());
    }

    public String getTitle() {
        return null;
//        Log.v(TAG, "app is " + service.getInfo().getApplication());
//        if (serviceLookup != null) {
//            return serviceLookup.getDescription();
//        }
//        return service.getType();
    }

    public String getSubTitle() {
        return null;
//        return service.getName();
    }

    public int getImageResource() {
        if (mServiceLookup != null) {
            return mServiceLookup.getDrawable();
        }
        return 0;
    }

}

File: flame/src/main/java/org/movieos/flame/models/FlameService.kt
package org.movieos.flame.models

import android.net.nsd.NsdServiceInfo
import org.movieos.flame.utilities.ServiceLookup
import java.util.*

class FlameService(s: NsdServiceInfo?) {

    var info: NsdServiceInfo
        internal set
    var hostIdentifiers: ArrayList<String>
        internal set
    var serviceLookup: ServiceLookup? = null
        internal set

    init {
        if (s == null) {
            throw RuntimeException("service must be set")
        }
        info = s
        hostIdentifiers = ArrayList<String>()
        //        if (service.getInfo() != null) {
        //            if (!service.getInfo().getServer().isEmpty()) {
        //                hostIdentifiers.add(service.getInfo().getServer());
        //            }
        //            for (String address : service.getInfo().getHostAddresses()) {
        //                if (!address.isEmpty()) {
        //                    hostIdentifiers.add(address);
        //                }
        //            }
        //            hostIdentifiers.add(toString());
        //
        //            serviceLookup = ServiceLookup.get(service.getInfo().getApplication());
        //        }
    }

    override fun equals(o: Any?): Boolean {
        if (o !is FlameService) {
            return false
        }
        return o.toString() == toString()
    }

    override fun toString(): String {
        return "${info.serviceName} ${info.serviceType}"
    }

    val ipAddress: String
        get() = info.host.hostAddress

    // TODO probably not good enough
    val identifier: String
        get() = "${info.serviceName} ${info.serviceType}"

    //        Log.v(TAG, "app is " + service.getInfo().getApplication());
    //        if (serviceLookup != null) {
    //            return serviceLookup.getDescription();
    //        }
    //        return service.getType();
    val title: String?
        get() = null

    //        return service.getName();
    val subTitle: String?
        get() = null

    val imageResource: Int
        get() {
            if (serviceLookup != null) {
                return serviceLookup!!.drawable
            }
            return 0
        }

}
